Programs/_testembed.c

```diff
@@ -1181,10 +1222,14 @@ static int test_audit_subinterpreter(void)
     PySys_AddAuditHook(_audit_subinterpreter_hook, NULL);
     _testembed_Py_Initialize();
 
+    PyThreadState *mainstate = PyThreadState_Get();
```

Comment:
You may want to double-check with @zooba on his intention here.  It's pretty important to make sure that the auditing functionality works as expected.

Reply:
Hi @zooba, as a consequence of my changes here, `test_audit_subinterpreter()` in `_testembed.c` started failing.

The change I'm making is to make `Py_Finalize()` implicitly clean up subinterpreters.

In the test, multiple subinterpreters are created, and then `Py_Finalize()` is called from the last-created subinterpreter. It seems there's currently an issue with calling `Py_Finalize()` from a subinterpreter (see bpo-37776), which caused this test to fail when getting `Py_Finalize()` to clean up subinterpreters.

The test passes if `Py_Finalize()` is instead called from the main interpreter tstate - which is the change I've made to the test. Just wanting to check whether that's taking anything away from what's intentionally being checked by this test?

Reply:
I also confirmed that this test still fails on my branch without this change.